modules/bullet/bullet_physics_server.h

```diff
@@ -51,44 +51,61 @@ class BulletPhysicsServer3D : public PhysicsServer3D {
 	friend class BulletPhysicsDirectSpaceState;
 
 	bool active = true;
-	char active_spaces_count = 0;
-	Vector<SpaceBullet *> active_spaces;
+	bool using_threads = false;
+	bool doing_sync;
 
-	mutable RID_PtrOwner<SpaceBullet> space_owner;
-	mutable RID_PtrOwner<ShapeBullet> shape_owner;
-	mutable RID_PtrOwner<AreaBullet> area_owner;
-	mutable RID_PtrOwner<RigidBodyBullet> rigid_body_owner;
-	mutable RID_PtrOwner<SoftBodyBullet> soft_body_owner;
-	mutable RID_PtrOwner<JointBullet> joint_owner;
+	Set<SpaceBullet *> active_spaces;
+
+	mutable RID_PtrOwner<SpaceBullet, true> space_owner;
+	mutable RID_PtrOwner<ShapeBullet, true> shape_owner;
+	mutable RID_PtrOwner<AreaBullet, true> area_owner;
+	mutable RID_PtrOwner<RigidBodyBullet, true> rigid_body_owner;
+	mutable RID_PtrOwner<SoftBodyBullet, true> soft_body_owner;
+	mutable RID_PtrOwner<JointBullet, true> joint_owner;
+	static BulletPhysicsServer3D *singletonbullet;
```

Comment:
Usually we implement the singleton pattern with a member named `singleton` and a public `get_singleton()` static method.

Though what changed recently that requires using a singleton instead of relying on `get_physics_server()`?

Reply:
Same think like above. I just was try to make it more uniform to the default physics classes. For sure it's not necessary.

![image](https://user-images.githubusercontent.com/76991284/115796571-589bd780-a3d2-11eb-84ac-15672bc8f5ff.png)

Reply:
![image](https://user-images.githubusercontent.com/76991284/115796609-69e4e400-a3d2-11eb-9dee-0381a6b36416.png)

Reply:
`get_physics_server` is actually the equivalent of a `BulletPhysicsServer3D` singleton, so there's no need for this change.